### DISCUSSION TITLE
Why is my -composite image turning out blurred?

### DISCUSSION THREAD

**Discussion by SentientBeingOnAQuest** [ImageMagick/ImageMagick#4589] [Help] [answered]
I am using the following command:

`for %F in (*.jpg) do magick ^
 ( "%F" +write mpr:img -blur 10x10 -resize x600 -gravity center -crop 600x600+0+0 +repage ) ^
 ( "%F" +write mpr:img -bordercolor Black -border 2x2 +repage ) ^
 mpr:img -gravity center -compose over -composite ^
 "result_%F"`

To achieve this: 
![result](https://user-images.githubusercontent.com/95970582/145889389-c39d7b17-b13d-48b3-b1f1-1d1eb220419d.jpg)

However what I want to achieve is this:
![Desired Result](https://user-images.githubusercontent.com/95970582/145890340-f5b07501-9886-4c67-9e1e-4aa66657784e.jpg)

The middle image and its 2x2 border are supposed to be completely 100x visible.
Can anyone help me figure out which part of my command is making the middle image blurry? 
I'm on Windows 10 IM7

**Comment by snibgo** [comment, marked as answer]
> Can anyone help me figure out which part of my command is making the middle image blurry?

Hmm. Tricky. I suspect "-blur 10x10" is blurring your image.

But I can't test it because you haven't shown your input image. You are doing a masked composite, using the input image as the mask, so we can expect to see some of the blurred image in the centre.

**Reply by fmw42** [reply]
I believe snibgo is correct in that it due to your using the input image as a mask in the composite. You also read the input twice and put the same input into mpr:img, which is redundant.

Try this

```
magick ^
logo: +write mpr:img +delete ^
( mpr:img -blur 10x10 -resize x600 -gravity center -crop 600x600+0+0 +repage ) ^
( mpr:img -bordercolor Black -border 2x2 +repage ) ^
-gravity center -compose over -composite ^
result.png

```

![result](https://user-images.githubusercontent.com/13458218/145907422-25f8cdb6-555d-4d2f-aaff-dade23741b47.png)

If this is what you want, then replace logo: with your actual input image and put it back into the loop.

Or if your image is wider than 600, then crop and shave to account for the black border

```
magick ^
logo: +write mpr:img +delete \
( mpr:img -blur 10x10 -resize x600 -gravity center -crop 600x600+0+0 +repage ) ^
( mpr:img -crop 600x600+0+0 +repage -shave 2x2 -bordercolor Black -border 2x2 +repage ) ^
-gravity center -compose over -composite ^
result.png

```

![result](https://user-images.githubusercontent.com/13458218/145907876-3069c6be-a65e-4a9f-a79c-a377ab580d6d.png)

**Reply by SentientBeingOnAQuest** [reply]
@fmw42 You solved the issue! Thank you! I edited the command to include the for loop:

`for %F in (*.jpg) do magick "%F" ^
 +write mpr:img +delete ^
 ( mpr:img -blur 10x10 -resize x600 -gravity center -crop 600x600+0+0 +repage ) ^
 ( mpr:img -bordercolor Black -border 2x2 +repage ) ^
 -gravity center -compose over -composite ^
 "result_%F"`

This works perfectly thank you!

**Comment by fmw42** [comment]
It is blurring due to your using the input image as a mask in the composite

**Reply by SentientBeingOnAQuest** [reply]
Your other comment solved the issue, thank you! If you repost it as an answer rather than a reply then I can mark your post as the answer.

**Reply by fmw42** [reply]
Actually snibgo was the first to suggest this. So he deserves the credit. I only tried to emphasize his point.

**Reply by SentientBeingOnAQuest** [reply]
Just another quick question. I suspect I'm having the same issue again but with text this time. I thought you might be able to help me. Why is the blur (in bold cursive) applying to the whole image rather than just the text in this code?:

for %F in (*.jpg) do magick "%F" ^
-resize 572x ^
+write mpr:img +delete ^
( mpr:img -blur 10x10 -resize x592 -gravity center -crop 592x592+0+0 +repage ) ^
( mpr:img -bordercolor #DB9D00 -border 1x1 -bordercolor #FDF9DC -border 1x1 -bordercolor #DB9D00 -border 1x1 +repage ) ^
-gravity center -compose over -composite ^
-bordercolor #DB9D00 -border 2x2 -bordercolor #FDF9DC -border 2x2 -bordercolor #DB9D00 -border 2x2 ^
-font San-Diego -pointsize 70 -stroke black -strokewidth 8 -annotate +25+65 "MARLINGBAITS" **_-blur 0x8_** ^
-fill white -stroke none -annotate +25+65 "MARLINGBAITS" ^
"result_%F"

(Win10 IM7)

**Comment by snibgo** [comment]
"-blur", like other operations such as "-border", operates on all the images in the current list. At that point in the command, you have only one image. So it blurs that image.

**Reply by SentientBeingOnAQuest** [reply]
(I assume you're referring to my reply to fmw42 and not my original post) So how can I isolate the blur to just the text? Putting parenthesis seems to have no effect.

for %F in (*.jpg) do magick "%F" ^
-resize 572x ^
+write mpr:img +delete ^
( mpr:img -blur 10x10 -resize x592 -gravity center -crop 592x592+0+0 +repage ) ^
( mpr:img -bordercolor #DB9D00 -border 1x1 -bordercolor #FDF9DC -border 1x1 -bordercolor #DB9D00 -border 1x1 +repage ) ^
-gravity center -compose over -composite ^
-bordercolor #DB9D00 -border 2x2 -bordercolor #FDF9DC -border 2x2 -bordercolor #DB9D00 -border 2x2 ^
-font San-Diego -pointsize 70 -stroke black -strokewidth 8 -annotate +25+65 "MARLINGBAITS" **_-blur 0x8_** ^
-fill white -stroke none -annotate +25+65 "MARLINGBAITS" ^
"result_%F"

**Comment by fmw42** [comment]
You only have one image at the time you are doing blurring. You are adding text to that image, then blurring, so the blur will affect the text and the image. You should create a new image of text on a transparent background. Blur that. Then composite that image over your background image.  See label: or caption: rather than -annotate.  See https://legacy.imagemagick.org/Usage/text/
